pkgs/development/tools/build-managers/scala-cli/update.nix

```diff
@@ -0,0 +1,37 @@
+{ lib, curl, writeShellScript, jq, gnused, git, nix, coreutils }: { platforms, pname, version }:
+
+writeShellScript "${pname}-update-script" ''
+  set -o errexit
+  PATH=${lib.makeBinPath [ curl jq gnused git nix coreutils ]}
+
+  latest_version=$(curl -s "https://api.github.com/repos/VirtusLab/scala-cli/releases?per_page=1" | jq ".[0].tag_name" --raw-output | sed 's/^v//')
+
+  if [[ "${version}" = "$latest_version" ]]; then
+      echo "The new version same as the old version."
+      exit 0
+  fi
+
+  nixpkgs=$(git rev-parse --show-toplevel)
+  sources_json="$nixpkgs/pkgs/development/tools/build-managers/scala-cli/sources.json"
+
+  platform_assets=()
+
+  for platform in ${lib.concatStringsSep " " platforms}; do
+    asset=$(jq ".assets.\"$platform\".asset" --raw-output < $sources_json)
+    release_asset_url="https://github.com/Virtuslab/scala-cli/releases/download/v$latest_version/$asset"
+
+    asset_hash=$(nix-prefetch-url "$release_asset_url")
+
+    asset_object=$(jq --compact-output --null-input \
+      --arg asset "$asset" \
+      --arg sha256 "$asset_hash" \
+      --arg platform "$platform" \
+      '{asset: $asset, sha256: $sha256, platform: $platform}')
+    platform_assets+=($asset_object)
+  done
+
+  printf '%s\n' "''${platform_assets[@]}" | \
+    jq -s "map ( { (.platform): . | del(.platform) }) | add" | \
+    jq --arg version $latest_version \
+      '{ version: $version, assets: . }' > $sources_json
+''
```

Comment:
do you really want to make this script writable to store? I think maybe using script with nix shebang would be better option for this. For instance this is what we use to update elm https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/elm/update.sh

Reply:
I don't mind really, I saw e.g. Sublime's: https://github.com/NixOS/nixpkgs/blob/c10e14acc786808d4bd07aabcbe0adc15af13ac0/pkgs/applications/editors/sublime/4/common.nix#L122-L139

Are there any downsides to this, aside from this being rebuilt after every update?

Reply:
well I think running a plain shell script is a bit simpler. also I assume this script is not really useful outside of nixpkgs maintenance. But there are no rules for this as far as I know. You can keep this as is. I would personally create shell script that is using nix-shell myself but ultimately you'll be one working with this not me.

Reply:
it's fine by me ;)